container/Dockerfile

```diff
@@ -0,0 +1,111 @@
+# BUILDER
+FROM alpine:latest AS builder
+
+#build ARGs
+ARG _RADARE2_GIT_REPO="https://github.com/radareorg/radare2.git"
+ENV RADARE2_GIT_REPO=$_RADARE2_GIT_REPO
+
+ARG _KS_NAV_GIT_REPO="https://github.com/elisa-tech/ks-nav"
+ENV KS_NAV_GIT_REPO=$_KS_NAV_GIT_REPO
+ARG _KS_NAV_GIT_REPO_BRANCH=main
+ENV KS_NAV_GIT_REPO_BRANCH=$_KS_NAV_GIT_REPO_BRANCH
+
+# install build dependecies
+RUN apk add --no-cache autoconf automake build-base git
+RUN apk add --no-cache go
+RUN apk add --no-cache linux-headers
+
+#build app
+WORKDIR /build
+
+# Build radare2
+RUN git clone ${RADARE2_GIT_REPO} /build/radare2 && cd /build/radare2 && git checkout ecd8d00aad289abf8f850b45d5f6c2e6b0c15c0f
+RUN cd /build/radare2 && ./configure && make && make install
+
+# Build ks-nav
+RUN git clone ${KS_NAV_GIT_REPO} /build/ksnav && cd /build/ksnav/kern_bin_db && git checkout ${KS_NAV_GIT_REPO_BRANCH}
+RUN cd /build/ksnav/kern_bin_db && make && make install
+RUN cd /build/ksnav/nav && make && make install
+RUN cd /build/ksnav/navweb && go get -u github.com/go-bindata/go-bindata/... && go install github.com/go-bindata/go-bindata/... && make && make install
+
+
+
+##############################################################################################################################################
+
+# RUNTIME
+FROM alpine:latest
+
+#run ARGs
+ARG _POSTGRES_NAME=only
+ENV POSTGRES_NAME=$_POSTGRES_NAME
+ARG _POSTGRES_DATA_DIR=/var/lib/postgresql/data/
+ENV POSTGRES_DATA_DIR=$_POSTGRES_DATA_DIR
+ARG _POSTGRES_IDENT_FILE=pg_ident.conf
+ENV POSTGRES_IDENT_FILE=$_POSTGRES_IDENT_FILE
+ARG _POSTGRES_CONF_FILE=${POSTGRES_DATA_DIR}/${POSTGRES_NAME}/postgresql.conf
+ENV POSTGRES_CONF_FILE=$_POSTGRES_CONF_FILE
+
+ARG _START_SCRIPT_NAME=/usr/bin/start.sh
+ENV START_SCRIPT_NAME=$_START_SCRIPT_NAME
+
+# install dependencies
+RUN apk add --no-cache postgresql postgresql-contrib postgresql-client
```

Comment:
Postgres is a very big gun - was sqlite also considered a candidate?

Reply:
SQLite3 holds significant potential. 
Firstly, it can be compiled statically, marking a substantial stride towards my original goal of achieving a distribution-free container. 
Its simpler configuration greatly streamlines the mechanics within the container. 
Moreover, its lightweight nature is another advantage. 
Regrettably, based on the tests conducted thus far (albeit not extensively), I have observed a decline in performance for tasks such as creating new databases and utilizing them to generate diagrams.

Before I can consider replacing PostgreSQL with SQLite3, I must identify a potential solution, if one exists, for this performance degradation. 
This entails scheduling a more comprehensive testing session.

In practical terms, two primary factors have deterred me from adopting SQLite3 in the current stage:
1. **Limited Testing**: SQLite3 has received minimal testing within this application. Its compatibility and performance are only marginally confirmed, in contrast to the more robustly validated PostgreSQL.
2. **Performance Discrepancy**: Notably, SQLite3 exhibits noticeably slower performance when compared to PostgreSQL, posing a hurdle to its immediate adoption.

With these challenges in mind, further exploration and analysis are required before SQLite3 can be considered a viable replacement for PostgreSQL.